packages/vscode-extension/src/envTree.ts

```diff
@@ -121,42 +149,85 @@ async function localSettingsExists(projectRoot: string): Promise<boolean> {
+    if (subscriptionInfo) {
+      const subscriptionTreeItem: TreeItem = {
+        commandId: `fx-extension.environment.${env}.subscription`,
+        label: subscriptionInfo.subscriptionName,
+        icon: "key",
+        isCustom: false,
+        parent: "fx-extension.environment." + env,
+      };
+
+      envSubItems.push(subscriptionTreeItem);
+
+      const resourceGroupName = await getResourceGroupNameFromEnv(env);
+      if (resourceGroupName) {
+        const resourceGroupTreeItem: TreeItem = {
+          commandId: `fx-extension.environment.${env}.resourceGroup`,
+          label: resourceGroupName,
+          icon: "symbol-method",
+          isCustom: false,
+          parent: `fx-extension.environment.${env}.subscription`,
+        };
 
-  const parentCommand = environmentTreeProvider.findCommand("fx-extension.environment." + env);
-  if (!parentCommand) {
-    return;
-  }
+        envSubItems.push(resourceGroupTreeItem);
+      }
 
-  const envSubItems: TreeItem[] = [];
-  const subscriptionInfo = await getSubscriptionInfoFromEnv(env);
-  if (subscriptionInfo) {
-    const subscriptionTreeItem: TreeItem = {
-      commandId: `fx-extension.environment.${env}.subscription`,
-      label: subscriptionInfo.subscriptionName,
-      icon: "key",
-      isCustom: false,
-      parent: "fx-extension.environment." + env,
-    };
+      const warningItem = await checkSubscriptionPermission(env, subscriptionInfo.subscriptionId);
+      if (warningItem) {
+        envSubItems = [warningItem].concat(envSubItems);
+      }
+    }
 
-    envSubItems.push(subscriptionTreeItem);
+    return envSubItems;
   }
 
-  const resourceGroupName = await getResourceGroupNameFromEnv(env);
-  if (resourceGroupName) {
-    const resourceGroupTreeItem: TreeItem = {
-      commandId: `fx-extension.environment.${env}.resourceGroup`,
-      label: resourceGroupName,
-      icon: "symbol-method",
-      isCustom: false,
-      parent: `fx-extension.environment.${env}.subscription`,
+  return [];
+}
+
+function checkAzureAccountStatus(env: string): TreeItem | undefined {
+  if (AzureAccountManager.getAccountInfo() === undefined) {
+    return {
+      commandId: `fx-extension.environment.${env}.checkAzureAccount`,
+      label: StringResources.vsc.commandsTreeViewProvider.noAzureAccountSignedIn,
+      icon: "warning",
+      isCustom: true,
+      parent: `fx-extension.environment.${env}`,
     };
+  } else {
+    return undefined;
+  }
+}
 
-    envSubItems.push(resourceGroupTreeItem);
+async function checkSubscriptionPermission(
+  env: string,
+  subscriptionId: string
+): Promise<TreeItem | undefined> {
+  if (tools.tokenProvider.azureAccountProvider.getAccountInfo()) {
+    const subscriptions: SubscriptionInfo[] | undefined =
+      await tools.tokenProvider.azureAccountProvider.listSubscriptions();
+
+    const targetSub = subscriptions.find((sub) => sub.subscriptionId === subscriptionId);
```

Comment:
Need to check if subscriptions is undefined here?

Reply:
Good catch. Will fix.